VEDA_OS01/transcripts.py

```diff
@@ -0,0 +1,267 @@
+            process_metadata.save()
+            LOGGER.exception(
+                '[CIELO24 TRANSCRIPTS] Fetch request failed for video=%s -- lang=%s -- job_id=%s',
+                video_id,
+                lang_code,
+                job_id
+            )
+            return
+
+        process_metadata.status = TranscriptStatus.READY
+        process_metadata.save()
+
+        try:
+            sjson = convert_srt_to_sjson(srt_data)
+            sjson_file_name = upload_sjson_to_s3(CONFIG, sjson)
+        except Exception:
+            LOGGER.exception(
+                '[CIELO24 TRANSCRIPTS] Request failed for video=%s -- lang=%s -- job_id=%s -- message=%s',
+                video_id,
+                lang_code,
+                job_id
+            )
+            raise
+
+        # update edx-val with completed transcript information
+        val_api = VALAPICall(process_metadata.video, val_status=None)
+        val_api.update_val_transcript(
+            video_id=process_metadata.video.studio_id,
+            lang_code=lang_code,
+            name=sjson_file_name,
+            transcript_format=TRANSCRIPT_SJSON,
+            provider=TranscriptProvider.CIELO24
+        )
+
+        # update transcript status for video in edx-val only if all langauge transcripts are ready
+        video_jobs = TranscriptProcessMetadata.objects.filter(video__studio_id=video_id)
+        if all(video_job.status == TranscriptStatus.READY for video_job in video_jobs):
+            val_api.update_video_status(process_metadata.video.studio_id, VideoStatus.TRANSCRIPTION_READY)
+
+
+def fetch_srt_data(url, **request_params):
+    """
+    Fetch srt data from transcript provider.
+    """
+    # return TRANSCRIPT_SRT_DATA
+    response = requests.get(
+        utils.build_url(url, **request_params)
+    )
+
+    if not response.ok:
+        raise TranscriptFetchError(
+            '[TRANSCRIPT FETCH ERROR] status={} -- text={}'.format(
+                response.status_code,
+                response.text
+            )
+        )
+
+    return response.text
+
+
+def convert_srt_to_sjson(srt_data):
+    """
+    Convert SRT to SJSON
+
+    Arguments:
+        srt_data: unicode, content of source subs.
+
+    Returns:
+        dict: SJSON data
+    """
+    srt_subs_obj = SubRipFile.from_string(srt_data)
+
+    sub_starts = []
+    sub_ends = []
+    sub_texts = []
+
+    for sub in srt_subs_obj:
+        sub_starts.append(sub.start.ordinal)
+        sub_ends.append(sub.end.ordinal)
+        sub_texts.append(sub.text.replace('\n', ' '))
+
+    subs = {
+        'start': sub_starts,
+        'end': sub_ends,
+        'text': sub_texts
+    }
+
+    return subs
+
+
+def upload_sjson_to_s3(config, sjson_data):
+    """
+    Upload sjson data to s3.
+    """
+    s3_conn = boto.connect_s3(
+        config['transcript_bucket_access_key'],
+        config['transcript_bucket_secret_key']
+    )
+    bucket = s3_conn.get_bucket(config['transcript_bucket_name'])
```

Comment:
@muhammad-ammar most of the buckets are not using access & secret key, can we also change this accordingly ?